lib/internal/modules/helpers.js

```diff
@@ -392,6 +392,24 @@ ObjectFreeze(compileCacheStatus);
 const constants = { __proto__: null, compileCacheStatus };
 ObjectFreeze(constants);
 
+async function resolveLoadAndCache(specifier, base = 'data:', importAttributes = kEmptyObject, conditions) {
+  const cascadedLoader = require('internal/modules/esm/loader').getOrInitializeCascadedLoader();
+  // TODO: this should hit the cache (and populate it if it finds nothing)
+  const { url, format: resolveFormat } = await cascadedLoader.resolve(`${specifier}`, `${base}`, importAttributes);
+  const { format, source } = await cascadedLoader.load(url, {
+    __proto__: null,
+    importAttributes,
+    format: resolveFormat,
+    conditions,
+  });
+  return {
+    __proto__: null,
+    format,
```

Comment:
When `--experimental-strip-types` is enabled and performing this operation on a `.ts` file, this `format` may contains `commonjs-typescript` and `module-typescript` which are not documented at https://nodejs.org/api/module.html#loadurl-context-nextload as final formats.

I think we should convert `commonjs-typescript` and `module-typescript` to `commonjs` and `module` respectively. Alternatively, document the other two and make them public.